device/CMakeLists.txt

```diff
@@ -41,16 +41,15 @@ target_include_directories(
 )
 
 # flatbuffers is public - exposed to tt_metal by tt_simulation_device_generated.h
-# nng is public - exposed to tt_metal by tt_simulation_host.hpp
 target_link_libraries(
     device
     PUBLIC
-        nng
```

Comment:
@blozano-tt Please make sure that tt_metal CI passes with this change. Copying the comment from the revert PR:

This change broke tt-metal build.
post commit workflow failing: https://github.com/tenstorrent/tt-metal/actions/runs/12031721201
post commit workflow on this branch: https://github.com/tenstorrent/tt-metal/actions/runs/12032379476/job/33545486354
blackhole post-commit failing: https://github.com/tenstorrent/tt-metal/actions/runs/12031811999
blackhole post-commit on this branch: https://github.com/tenstorrent/tt-metal/actions/runs/12032509241

Reply:
Issues seem to be fixed here:


https://github.com/tenstorrent/tt-metal/actions/runs/12113533333

Reply:
https://github.com/tenstorrent/tt-metal/issues/15527#issuecomment-2510682445

Reply:
Thanks for fixing this!